group_project/group_activity.py

```diff
@@ -1,3 +1,75 @@
+"""
+This module parses Group project XML into a tree of domain-specific objects.
+Basically it looks like the following
+
+GroupActivity (.)
+- ActivityComponent (./projectcomponent)
+-- ActivitySection (//section - different paths)
+--- ActivityQuestion (//section/question)
+---- <no dedicated class, etree.Element is used> (//section/question/answer)
+--- ActivityAssessment (//section/assessment)
+---- <no dedicated class, etree.Element is used> (//section/assessment/answer)
+
+GroupActivity (paths relative to root element)
+- resources: [DottableDict(title, description, location)] - ./resources/document
+- grading_criteria: [DottableDict(title, description, location)] - ./resources/document[@grading_criteria="true"]     # attribute filter implicit
+- milestone_dates: {str: datetime.date} - ./dates/milestone
+- submissions: [DottableDict(id, title, description, location?)] - ./submissions/document
+- grade_questions: [ActivityQuestion] - //section/question
+- activity_components: [ActivityComponent] - ./projectcomponent
+- grading_override: Bool                                                                # if True - allows visiting components after close date; used by TA
+* has_submissions: Bool                                                                 # True if ANY submission uploaded
+* has_all_submissions: Bool                                                             # True if ALL submission uploaded
+* submission_json: json                                                                 # submissions serialized into json format
+* step_map: json
+    {
+        component_id: { prev: prev_component.id, name: component.name, next: next_component.id},
+        ordered_list: [component1.id, component2.id, ...],
+        default: <latest_open_component.id if not grading_override else latest_component_with_other_group_sections.id>
+    }
```

Comment:
Also, the order of children allows to determine this already, and if there was a need for next/prev this would be attributes on the individual Stage XBlocks (if the XBlock API doesn't already provide this out of the box, actually)